src/scripts/main.js

```diff
@@ -1,3 +1,315 @@
+    const [namePerson, position, office, age, salary] = body
+      .children[i].children;
+
+    people.push({
+      name: namePerson.innerText,
+      position: position.innerText,
+      office: office.innerText,
+      age: age.innerText,
+      salary: getNumber(salary.innerText),
+    });
+  }
+};
+
+const setClick = (list) => {
+  list.forEach(element => {
+    element.addEventListener('click', (e) => {
+      if (!textElement) {
+        textElement = e.target.innerText;
+      }
+
+      if (document.querySelector('.active')) {
+        document.querySelector('.active').classList.remove('active');
+      }
+
+      element.classList.add('active');
+
+      const [namePerson, position, office, age, salary] = element.children;
+
+      selectObj = {
+        name: namePerson.innerText,
+        position: position.innerText,
+        office: office.innerText,
+        age: age.innerText,
+        salary: getNumber(salary.innerText),
+      };
+    });
+  });
+};
+
+const createTr = () => {
+  const text = document.querySelector('.cell-input').value;
+  const td = document.querySelector('.cell-input').parentElement;
+
+  document.querySelector('.cell-input').parentElement.innerHTML = '';
+  td.innerText = !text ? textElement : text;
+  CreateObjectsByContent();
+};
+
+const setDobleClick = (list) => {
+  for (let i = 0; i < list.length; i++) {
+    list[i].addEventListener('dblclick', (e) => {
+      const input = document.createElement('input');
+
+      lostFocus(input);
+
+      if (document.querySelector('.cell-input')) {
+        createTr();
+      }
+
+      input.value = e.target.innerText;
+      input.classList.add('cell-input');
+      e.target.innerHTML = '';
+      e.target.appendChild(input);
+    });
+  }
+};
+
+const setKeyboard = (list) => {
+  for (let i = 0; i < list.length; i++) {
+    list[i].addEventListener('keypress', (e) => {
+      if (e.key === 'Enter' && e.target.value !== '') {
+        createTr();
+      } else if (e.key === 'Enter') {
+        const td = document.querySelector('.cell-input').parentElement;
+
+        document.querySelector('.cell-input').parentElement.innerHTML = '';
+        td.innerText = textElement;
+        textElement = '';
+      }
+    });
+  }
+};
+
+const lostFocus = (element) => {
+  element.addEventListener('blur', (e) => {
+    createTr();
+  });
+};
+
+const functionFoSort = (sortKey) => {
+  return (a, b) => {
+    const firstElement = a[sortKey];
+    const secondElement = b[sortKey];
+
+    if (flag === 'ASC') {
+      if (strings.includes(sortKey)) {
+        return firstElement.localeCompare(secondElement);
+      }
+
+      if (numbers.includes(sortKey)) {
+        return firstElement - secondElement;
+      }
+    } else {
+      if (strings.includes(sortKey)) {
+        return secondElement.localeCompare(firstElement);
+      }
+
+      if (numbers.includes(sortKey)) {
+        return secondElement - firstElement;
+      }
+    };
+  };
+};
+
+sortBy.forEach(el => {
```

Comment:
use [eventDelagation](https://javascript.info/event-delegation)

Reply:
not fixed
why do we need to add listener to each element when you can use event delegation